### DISCUSSION TITLE
ESMValCore v2.10.0 release

### DISCUSSION THREAD

**Discussion by zklaus** [ESMValGroup/ESMValCore#2222] [Releases]
Dear @ESMValGroup/esmvaltool-developmentteam, 

today we are kicking off the release process for release 2.10.0.
Over the next four weeks, we'll prepare the release of ESMValTool v2.10.0, starting with the feature freeze of ESMValCore today.
Stay tuned for further news and updates here.

<div align="center">

| Manager | Buddy | Feature Freeze | Feature branch | Milestone |
| :---: | :---: | :---: | :---: | :---: |
| @zklaus | @bouweandela | `2023-10-09` | [![v2.10.x](https://img.shields.io/badge/branch-v2.10.x-yellow?logo=github&logoColor=lightgrey)](https://github.com/ESMValGroup/ESMValCore/tree/v2.10.x) | [![GH Milestone](https://img.shields.io/badge/milestone-v2.10.0-yellow?logo=github&logoColor=lightgrey)](https://github.com/ESMValGroup/ESMValCore/milestone/16) |

</div>

<div align="center">

| Minor Release | Status | Date | RTD | Release | conda-forge | PyPI | Docker
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v2.10.0rc1` | [![status](https://img.shields.io/badge/v2.10.0rc1-released-green)](https://github.com/ESMValGroup/ESMValCore/discussions/2222) | `2023-11-01` | [Docs](https://docs.esmvaltool.org/projects/ESMValCore/en/v2.10.0rc1/quickstart/index.html) | [v2.10.0rc1](https://github.com/ESMValGroup/ESMValCore/releases/tag/v2.10.0rc1) | [esmvalcore_rc](https://anaconda.org/conda-forge/esmvalcore?version=2.10.0rc1&channel=esmvalcore_rc) | [esmvalcore (--pre)](https://pypi.org/project/esmvalcore/2.10.0rc1/) | [docker](https://hub.docker.com/layers/esmvalgroup/esmvalcore/v2.10.0rc1/images/sha256-c4a24e4c06932b409abcdc14f53ea3d9a43acb28fa563105dcd727de5354655f?context=explore) |
| `v2.10.0` | [![status](https://img.shields.io/badge/v2.10.0-released-green)](https://github.com/ESMValGroup/ESMValCore/discussions/2222) | `2023-12-19` | [Docs](https://docs.esmvaltool.org/projects/ESMValCore/en/v2.10.0/quickstart/index.html) | [v2.10.0](https://github.com/ESMValGroup/ESMValCore/releases/tag/v2.10.0) | [esmvalcore](https://anaconda.org/conda-forge/esmvalcore?version=2.10.0) | [esmvalcore](https://pypi.org/project/esmvalcore/2.10.0/) | [docker](https://hub.docker.com/layers/esmvalgroup/esmvalcore/v2.10.0/images/sha256-?context=explore) |

</div>

**Comment by valeriupredoi** [comment]
Almost 3:10 to Yuma :grin: (great movie BTW)
![310toYuma](https://github.com/ESMValGroup/ESMValCore/assets/28983971/373362f9-c55d-49d5-bc2d-c5b076941358)

**Comment by zklaus** [comment]
As a sanity check before making the release candidate, I ran the following recipes. A check means the recipe ran successfully.

- [x] bock20jgr/recipe_bock20jgr_fig_1-4.yml (#2239)
- [x] clouds/recipe_clouds_bias.yml
- [ ] clouds/recipe_lauer22jclim_fig3-4_zonal.yml (Ran out of time)
- [x] examples/recipe_decadal.yml
- [x] examples/recipe_python.yml
- [x] examples/recipe_ncl.yml
- [x] examples/recipe_r.yml
- [ ] ipccwg1ar6ch3/recipe_ipccwg1ar6ch3_atmosphere.yml (#2237)
- [x] ipccwg1ar5ch9/recipe_flato13ipcc_figure_945a.yml
- [ ] recipe_anav13jclim.yml (Data problem on test machine, possibly #2239)
- [x] recipe_autoassess_landsurface_soilmoisture.yml
- [x] recipe_autoassess_stratosphere.yml
- [x] recipe_climwip_test_basic.yml (#2237)
- [x] recipe_climwip_test_performance_sigma.yml (#2237)
- [ ] recipe_collins13ipcc.yml (#2239)
- [x] recipe_cmug_h2o.yml
- [x] recipe_cox18nature.yml
- [x] recipe_extreme_events.yml
- [x] recipe_extreme_index.yml
- [x] recipe_impact.yml (#2237)
- [x] recipe_hyint.yml
- [x] recipe_meehl20sciadv.yml
- [x] recipe_toymodel.yml

**Reply by bouweandela** [reply]
If you're using the distributed scheduler, the failure of clouds/recipe_lauer22jclim_fig3-4_zonal.yml may be related to https://github.com/SciTools/iris/issues/5457

**Reply by zklaus** [reply]
I did not use the distributed scheduler, but best I can tell, that recipe simply ran out of time; nothing else wrong, so I'll defer further testing to after the rc.

**Reply by zklaus** [reply]
I have unchecked `recipe_bock20jgr_fig_1-4.yml` now because I think it might be linked to #2239. I had previously fixed the version problem by adding `version` to the filename template for `obs4MIPs` in `config_developer.yml`, but seeing as the version problem is a wider issue, I would like to address that first and see if it solves the `obs4MIPs` issue as well.

**Comment by bouweandela** [comment]
@ESMValGroup/esmvaltool-developmentteam Would anyone have suggestions for highlights for the ESMValCore v2.10 changelog? We're planning to make the release today. [Here](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/changelog.html#id15) is an example of what that looked like for v2.9.

**Reply by schlunma** [reply]
- Added percentile operation for statistics preprocessor (https://github.com/ESMValGroup/ESMValCore/pull/2191)
- Improved output website (https://github.com/ESMValGroup/ESMValCore/pull/2062)
- Further progress regarding out-of-core computing/dask/memory efficiency: https://github.com/ESMValGroup/ESMValCore/pull/2205, https://github.com/ESMValGroup/ESMValCore/pull/2200, https://github.com/ESMValGroup/ESMValCore/pull/2169, https://github.com/ESMValGroup/ESMValCore/pull/2146

**Comment by axel-lauer** [comment]
This is one of my highlights (but probably rather v2.11?):

* https://github.com/ESMValGroup/ESMValCore/pull/2258

**Comment by bouweandela** [comment]
The release is now available! :partying_face: A big thanks to everyone who contributed! :star2:
